exceptions/pulp/exceptions/base.py

```diff
@@ -11,19 +11,20 @@ class PulpException(Exception):
     """
     http_status_code = http.client.INTERNAL_SERVER_ERROR
 
-    def __init__(self, *args):
-        super(PulpException, self).__init__(*args)
-        self.error_code = error_codes.PLP0000
-        self.error_data = {}
-
-        # child exceptions are those that are wrapped within this exception, validation errors
-        # for example would have one overall validation error and then a separate sub error
-        # for each validation that failed
-
-        self.child_exceptions = []
-
-    def add_child_exception(self, exception):
-        self.child_exceptions.append(exception)
+    def __init__(self, error_code=None, **kwargs):
```

Comment:
I imagined all `PulpException`'s are required to have codes. If someone subclasses this they should be required to specify a code. What do you think?